src/document/mod.rs

```diff
@@ -151,7 +152,9 @@ impl Document {
         let mut result = String::new();
         match code_blocks.get(&entrypoint) {
             Some(blocks) => {
-                for block in blocks {
+                for idx in 0..blocks.len() {
```

Comment:
`blocks.iter().enumerate()`?

Reply:
But don't I then get a problem to borrow indexed blocks? Probably not as long as I do no mutable borrows. I will try...